subscriptions/fixed-price/checkout/node/src/server.ts

```diff
@@ -27,55 +33,114 @@ const endpointSecret = process.env.DIAGONAL_WEBHOOK_ENDPOINT_SECRET as string
-app.post('/create-account/', async (req, res) => {
-  const email = req.body.email
-  // const name = req.body.name
-  // ...
+/**
+ * Plans
+ * Since you will be integrating fixed price subscriptions
+ * you will need some notion of plans/products, either defined locally, in your database,
+ * or using an external provider. In this example we choose to define plans locally.
+ */
+const basicPlan = {
+  id: '1',
+  amount: '10',
+  tokens: [Token.DAI, Token.USDC],
+  chains: [Chain.ETHEREUM, Chain.POLYGON],
+  interval: 'month' as const,
+  intervalCount: 1,
+}
 
-  // step 1: create diagonal customer
-  const customer = await diagonal.customers.create({
-    email,
-  })
+const premiumPlan = {
+  id: '2',
+  amount: '50',
+  tokens: [Token.DAI, Token.USDC],
+  chains: [Chain.ETHEREUM, Chain.POLYGON],
+  interval: 'month' as const,
+  intervalCount: 1,
+}
 
-  // step 2: create a user in your database, store Diagonal customer id along side it
-  // createUser(email, name, customer.id, ...)
+/**
+ * Checkout session
+ *
+ *  IMPORTANT fields
+ *  - `customer_id`: WHO is subscribing
+ *                   We recommend you pass a Diagonal customer id on checkout creation.
+ *                   `customer_id` will help you link a paying subscriber to a completed checkout session.
+ *
+ *  - `reference`  : WHAT they are paying for
+ *                   We recommend you pass a `reference` for what the user is paying for on checkout creation.
+ *                   Ideally this reference would refer to a unique product/plan/invoice id.
+ *
+ *
+ *  NOTE: `customer_id` and `reference` will be available in the `subscription.created` webhook event.
+ *         This will help you verify WHO subscribed and WHAT they paid for.
+ */
+app.post('/create-checkout-session/', async (req, res) => {
+  /* 
+     ***************************** Database code - rewrite yourself ********************************      
+      
+     Create a Diagonal `Customer` if not found in DB
 
-  res.sendStatus(200)
-})
+      ```
+        // Step 1: Get user from your DB
+        const user = UserTable.findOne({
+          req.user.id
+        })
 
-// Checkout sessions
-app.post('/create-checkout-session/', async (req, res) => {
-  // While creating a checkout session, you can pass in a customer ID
-  // to associate the session with a customer. This will allow you to
-  // retrieve the customer's subscriptions later.
-  let customerId = req.body.customerId
+        let diagonalCustomerId = user.diagonalCustomerId;
+
+        // Step 2: If user does not have a Diagonal customer id, create a new id and update user DB
+        if (diagonalCustomerId === undefined) {
+          
+          // Uniquely identify your customers
+          const diagonalCustomer = await diagonal.customers.create({
+            email: user.email,
+            name: user.name,
+            reference: user.id // optionally pass some unique ref if (email, name) is not unique
```

Comment:
The reference is not about uniqueness of name, email, but serves as a link with the local user. Maybe we can mention that?